src/hotspot/share/classfile/javaClasses.cpp

```diff
@@ -3014,9 +3014,10 @@ void java_lang_ClassFrameInfo::serialize_offsets(SerializeClosure* f) {
 
 static int get_flags(const methodHandle& m) {
   int flags = (jushort)( m->access_flags().as_short() & JVM_RECOGNIZED_METHOD_MODIFIERS );
-  if (m->is_initializer()) {
+  if (m->is_object_initializer()) {
     flags |= java_lang_invoke_MemberName::MN_IS_CONSTRUCTOR;
```

Comment:
No, it is not filtered, we still have `clinit`-s on this path. In the initial version https://github.com/openjdk/jdk/pull/20120/commits/6769cfe609849aa9ed0985dcbecb2b0aa24bca03 I caught the assert in many tests, mostly in stack traces generation. 

Yes, this changes the behavior: `clinit` would now be recorded as "method", instead of "constructor". Tracing back the uses of `get_flags`: it is used for initializing `java.lang.ClassFrameInfo.flags`. There seem to be no readers for this field in VM. Java side for `j.l.CFI` does not seem to check any method/constructor flags. So I would say this change in behavior is not really visible, and there is no need to try and keep the old (odd) behavior.

Reply:
Okay, such a change in behaviour was unexpected for a "cleanup" PR. I'm looking into it now. Perhaps @mlchung can comment?

Reply:
Yeah, this is not really a cleanup (behaviors stay the same) change. For this particular hunk, keeping the old behavior seems to be unnecessary work. Note that we are also changing the behavior in C2: both in `do_exits` we no longer emit the barriers for `static final` stores in `clinits`, plus EA does not care about `clinits` anymore as well. Those are also behavioral changes.

If you prefer, I can turn this PR into a behaviorally similar cleanup, and do the behavior changes separately.

Reply:
I certainly would not want those C2 changes to hidden behind what looks like a cleanup on the surface, so please do separate things out.

BTW Mandy is away for a while so we can't get her input on the original intent here.

Reply:
Fine by me. I am splitting out C2 parts here:
https://bugs.openjdk.org/browse/JDK-8336465
https://bugs.openjdk.org/browse/JDK-8336466

I'll probably fork `get_flags` change as a separate bug as well. 

This PR would then be only the final non-behavioral cleanups that would eliminate the remnant uses of `is_initializer`.

Reply:
> I'll probably fork get_flags change as a separate bug as well.

Now part of: https://bugs.openjdk.org/browse/JDK-8336468